tidy3d/components/structure.py

```diff
@@ -99,14 +101,24 @@ def eps_diagonal(self, frequency: float, coords: Coords) -> Tuple[complex, compl
     @pydantic.validator("medium", always=True)
     def _check_2d_geometry(cls, val, values):
         """Medium2D is only consistent with certain geometry types"""
+        geom = values.get("geometry")
         if isinstance(val, Medium2D):
             # validate that the geometry is actually 2d
-            geom = values.get("geometry")
             if not geom:
                 raise SetupError(
                     "Can't validate 2D structure because its geometry did not pass validation."
                 )
             _ = geom._normal_2dmaterial
+        elif geom is not None:
+            if np.count_nonzero(geom.bounding_box.size) < 3:
+                zero_axis = geom.bounding_box.size.index(0)
```

Comment:
note that `.index()` just returns the first index. if there are several dimensions of size zero, this will just return the first one. Could either change the warning message to make this more clear or use
```py
zero_axes = [ind for ind, ele in enumerate(geom.bounding_box.size) if ele == 0.0]
```